src/pkgdev/scripts/pkgdev_tatt.py

```diff
@@ -0,0 +1,261 @@
+    repo = namespace.domain.repos['gentoo'].raw_repo
+    return dict(match_package_list(repo, bug, only_new=True, filter_arch=[namespace.domain.arch])).keys()
+
+
+def _get_cmd_packages(namespace):
+    repos = namespace.domain.source_repos_raw
+    for pkgs in pkgutils.groupby_pkg(repos.itermatch(namespace.restrict, sorter=sorted)):
+        pkg = max(pkgs)
+        yield pkg.repo.match(pkg.versioned_atom)[0]
+
+
+def _groupby_use_expand(assignment: dict[str, bool], domain, pkg):
+    iuse: frozenset[str] = frozenset(pkg.iuse_stripped)
+    use_expands = tuple(s.lower() + '_' for s in domain.profile.use_expand)
+    use_expand_dict = defaultdict(set)
+
+    _immutable, enabled, _disabled = domain.get_package_use_unconfigured(pkg)
+
+    for var, state in assignment.items():
+        if var not in iuse:
+            continue
+        if state == (var in enabled):
+            continue
+        for use_expand in use_expands:
+            if var.startswith(use_expands):
+                if state:
+                    use_expand_dict[use_expand[:-1]].add(var.removeprefix(use_expand))
+                break
+        else:
+            use_expand_dict["USE"].add(('' if state else '-') + var)
+    return use_expand_dict
+
+
+def _build_job(namespace, pkg, is_test):
+    default_on_iuse = tuple(use[1:] for use in pkg.iuse if use.startswith('+'))
+    immutable, enabled, _disabled = namespace.domain.get_package_use_unconfigured(pkg)
+
+    iuse = frozenset(pkg.iuse_stripped)
+    force_true = immutable.union(('test', ) if is_test else ())
+    force_false = (('test', ) if not is_test else ())
+
+    if namespace.random_use == 'd':
+        prefer_true = enabled.union(default_on_iuse)
+    elif namespace.random_use in 'rR':
+        ignore_prefixes = set(namespace.ignore_prefixes)
+        if namespace.random_use == 'r':
+            ignore_prefixes.update(s.lower() + '_' for s in namespace.domain.profile.use_expand)
+        ignore_prefixes = tuple(ignore_prefixes)
+
+        prefer_true = [
+            use for use in iuse.difference(force_true, force_false)
+            if not use.startswith(ignore_prefixes)]
+        if prefer_true:
+            random.shuffle(prefer_true)
+            prefer_true = prefer_true[:random.randint(0, len(prefer_true) - 1)]
+        prefer_true.extend(use for use in enabled.union(default_on_iuse) if use.startswith(ignore_prefixes))
+
+    solutions = find_constraint_satisfaction(pkg.required_use, iuse.union(immutable),
+        force_true, force_false, frozenset(prefer_true))
+    for solution in solutions:
+        yield ' '.join(
+            f'{var.upper()}="{" ".join(vals)}"'
+            for var, vals in _groupby_use_expand(solution, namespace.domain, pkg).items()
+        )
+
+
+def _build_jobs(namespace, pkgs):
+    for pkg in pkgs:
+        if namespace.test and 'test' in pkg.defined_phases:
+            yield '='+pkg.cpvstr, True, next(iter(_build_job(namespace, pkg, True)))
```

Comment:
`pkg.versioned_atom` probably works btw, although if this is a multi-repo setup, neither form will pin the repo_id.